Source/JavaScriptCore/jit/AssemblyHelpers.cpp

```diff
@@ -1424,91 +1424,43 @@ void AssemblyHelpers::copyCalleeSavesToEntryFrameCalleeSavesBufferImpl(GPRReg ca
 #endif
 }
 
-void AssemblyHelpers::cageWithoutUntagging(Gigacage::Kind kind, GPRReg storage, bool mayBeNull)
+void AssemblyHelpers::cageWithoutUntagging(Gigacage::Kind kind, GPRReg storage)
 {
 #if GIGACAGE_ENABLED
     if (!Gigacage::isEnabled(kind))
         return;
-
-#if CPU(ARM64E)
-    RegisterID tempReg = InvalidGPRReg;
-    Jump skip;
-    if (kind == Gigacage::Primitive) {
-        if (mayBeNull)
-            skip = branchPtr(Equal, storage, TrustedImmPtr(JSArrayBufferView::nullVectorPtr()));
-        tempReg = getCachedMemoryTempRegisterIDAndInvalidate();
-        and64(TrustedImm64(Gigacage::mask(kind)), storage, tempReg);
-        addPtr(TrustedImmPtr(Gigacage::basePtr(kind)), tempReg);
-        // Flip the registers since bitFieldInsert only inserts into the low bits.
-        std::swap(storage, tempReg);
-    } else {
-        and64(TrustedImm64(Gigacage::mask(kind)), storage);
-        addPtr(TrustedImmPtr(Gigacage::basePtr(kind)), storage);
-    }
-    if (kind == Gigacage::Primitive)
-        insertBitField64(storage, TrustedImm32(0), TrustedImm32(64 - maxNumberOfAllowedPACBits), tempReg);
-    if (skip.isSet())
-        skip.link(this);
-#else
-    UNUSED_PARAM(mayBeNull);
     andPtr(TrustedImmPtr(Gigacage::mask(kind)), storage);
     addPtr(TrustedImmPtr(Gigacage::basePtr(kind)), storage);
-#endif
-
 #else
     UNUSED_PARAM(kind);
     UNUSED_PARAM(storage);
-    UNUSED_PARAM(mayBeNull);
 #endif
 }
 
 // length may be the same register as scratch.
-void AssemblyHelpers::cageConditionallyAndUntag(Gigacage::Kind kind, GPRReg storage, GPRReg length, GPRReg scratch, bool validateAuth, bool mayBeNull)
+void AssemblyHelpers::cageConditionallyAndUntag(Gigacage::Kind kind, GPRReg storage, GPRReg length, GPRReg scratch)
```

Comment:
You can remove the "AndUntag" part in the name now.  Ditto below.

Reply:
Done.